spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/AbstractConfigProperties.java

```diff
@@ -29,6 +29,8 @@
 
 	protected String namespace;
 
+	protected boolean useNameAsPrefix;
```

Comment:
As the documentation shows, I opted for two `useNameAsPrefix`:

 - `spring.cloud.kubernetes.config.useNameAsPrefix`
 - `spring.cloud.kubernetes.config.sources.useNameAsPrefix`

The second one can override the first one. imho, the documentation explains this. If not, I will be happy to provide more details.